docs/changelog/0.4.0.md

```diff
@@ -0,0 +1,532 @@
+
+
+## Sbt plugin changes
+### Native Config
+We've decided to deprecate the current spare set of `NativeX` setting keys with single `NativeConfig` object. 
+Existing 
+
+```scala
+// build.sbt
+nativeConfig ~= {
+  _.withMode(build.Mode.releaseFast)
+   .withGC(build.GC.immix)
+   .withLTO(build.LTO.full)
+   .withOptimize(true)
+   .withCompileOptions(Nil)
+   .withLinkingOptions(Nil)
+}
+```
+Old style settings keys are still respected, but with lower priority than the new config and will be removed at some point in the future.
+In the following example resulting `LTO` setting would be set to `thin`
+```scala
+nativeConfig := nativeConfig.value.withLTO(build.LTO.thin)
+
+nativeLTO := "none"
+```
+
+### Cross compilation
+With the latest release you're able to define a custom target for compiler by providing [LLVM style TargetTriple](https://clang.llvm.org/docs/CrossCompilation.html) in your config.
+This change does not change default behaviour of Scala Native sbt plugin, cousing to target host architecture and operating system.
+
+For example, if you're working on Linux and would like to create perfectly suited assembly for some specific version of MacOS without changing your whole build,
+you can use ad-hoc sbt setting:
+```bash
+sbt 'set nativeConfig ~= {_.withTargetTriple("x86_64-apple-darwin<version>")}' myApp/nativeLink
+```
+
+We consider changing target triple as a feature for advanced users, and cannot promise it would currently work with any possible configuration yet.
+However, the number of supported architectures and operating systems would definitely grow in the future.
+
+> When using Linux / MacOS you can check target triple used in your environment by command `llvm-config --host-target`.
+
+### Native sources in build
+With the 0.4.0 release you're able to put your C/C++ sources in the `resources/scala-native` directory inside your project,
+so they will be linked and compiled inside the SN pipeline.
+
+As an example you can use it to access macro defined constants and functions or to pass `struct`s from the stack to C functions.
+```c
+// src/resources/scala-native/example.c
+typedef int (*Callback0) (void);
+
+const int EXAMPLE_CONSTANT = 42;
+
+int exec(Callback0 f) {
+ return f();
+};
+
+```
+```scala
+// src/main/example.scala
+@extern
+object example {
+ def exec(cb: CFuncPtr0[CInt]): ExecResult = extern
+
+ @name("EXAMPLE_CONSTANT")
+ final val someConstant: Int = extern
+}
+```
+
+### Other
+* We've decided to no longer support sbt 0.13.x version - if your build still uses this legacy version you need to upgrade to version  1.1.6 or newer.
+* Failures during the build of multiple projects using common jar were fixed. 
+* The minimal version of Clang working with Scala Native is now 6.0
+* You can also observe lower overall memory usage when compiling and linking.
+
+## Native lib and unsafe package changes
+### CFuncPtr changes
```

Comment:
This should come first, along with other obviously breaking changes.

In general, the changelog should be structured along:

* Breaking changes
* New features
* Bug fixes

The categorizations along compiler/sbt-plugin/library are secondary to that.

Reply:
Also, the `CFuncPtr` changes are truly *language* changes at the core. They happened to be accompanied by library changes to support the language changes.